### ISSUE TITLE
Retina image version should be generated out of the box inside `srcset`

### ISSUE THREAD

**Issue by tomekrozalski** [Josh-McFarlin/remix-image#38]
### Describe the bug

We get incorrect `srcset` in HTML. It does not have retina width images.

### Steps to Reproduce the Bug or Issue

When we create an Image as below:

```
<Image
  alt=""
  responsive={[
    {
      maxWidth: 768,
      size: { width: 420 },
    },
    {
      size: { width: 800 },
    },
  ]}
  src="zzz.jpg"
/>
```

We get in HTML source:

```
<img
  alt=""
  src="/api/image?src=zzz&amp;width=800"
  srcset="
    /api/image?src=zzz&width=420 420w,
    /api/image?src=zzz&width=800 800w
  "
  sizes="(max-width: 768px) 420px, 800px"
/>
```

We have here 420-px-width and 800-px-width image inside `srcset`. Alright, but what about retina screens? :)

### Expected behavior

We should generate 2x size images in `srcset` out of the box, so the output should be:

```
<img
  alt=""
  src="/api/image?src=zzz&amp;width=800"
  srcset="
    /api/image?src=zzz&width=420 420w,
    /api/image?src=zzz&width=800 800w,
    /api/image?src=zzz&width=840 840w,
    /api/image?src=zzz&width=1600 1600w
  "
  sizes="(max-width: 768px) 420px, 800px"
/>
```

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

**Comment by Josh-McFarlin** [comment]
I like this proposal. I think the `Image` component could be given a new optional prop called something like `retina` that would.

```typescript
interface ImageProps {
  ... current props ...,
  retina?: number | number[];
}
```

`retina` would have a default of 1. If it is supplied a `number` such as `3`, it would be equivalent to `retina=[1, 3]` and would generate:
```
srcset="
  /api/image?src=zzz&width=420 420w,
  /api/image?src=zzz&width=1260 1260w
"
```

If it is supplied a `number[]` such as `[2,3]`, it will concat it with `[1]`, so it would be equivalent to `retina=[1,2,3]` and generate `srcset` for images with sizes `1x`, `2x`, and `3x`.

This is a pretty easy change to make, and can all be done from modifying the frontend `Image` component and the `useResponsiveImage` hook. I can implement these changes in the next few days. Any thoughts on this implementation?

**Comment by Josh-McFarlin** [comment]
@tomekrozalski Just released RemixImage v1.1.0 that includes the prop `dprVariants` accepting `number` or `number[]`. This value is always merged with the array `[1]`. Documentation here: https://remix-image.mcfarl.in/docs/component#proptypes
